includes/modules/payment/securesubmit.php

```diff
@@ -164,50 +174,42 @@ public function process_button() {
 
     public function before_process() {
         global $_POST, $order, $sendto, $currency, $charge, $db, $messageStack; 
-        require_once(DIR_FS_CATALOG . DIR_WS_MODULES . 'payment/securesubmit/Hps.php');
+        require 'vendor/autoload.php';
         $error = '';
         
-        $config = new HpsConfiguration();
-        $config->secretApiKey = MODULE_PAYMENT_SECURESUBMIT_SECRET_KEY;
-        $config->versionNumber = '1512';
-        $config->developerId = '002914';
+        $chargeService = $this->getConfig();
 
-        $chargeService = new HpsCreditService($config);
-
-        $hpsaddress = new HpsAddress();
+        $hpsaddress = new Address();
         $hpsaddress->address = $order->billing['street_address'];
         $hpsaddress->city = $order->billing['city'];
         $hpsaddress->state = $order->billing['state'];
         $hpsaddress->zip = preg_replace('/[^0-9]/', '', $order->billing['postcode']);
         $hpsaddress->country = $order->billing['country']['title'];
 
-        $cardHolder = new HpsCardHolder();
+        $cardHolder = new Customer();
         $cardHolder->firstName = $order->billing['firstname'];
         $cardHolder->lastName = $order->billing['lastname'];
         $cardHolder->emailAddress = $order->customer['email_address'];
         $cardHolder->address = $hpsaddress;
```

Comment:
The card holder name is now set on the `CreditCardData` object, and the address is passed via the `withAddress` method on the builder.

Reply:
updated